app/types/bool.go

```diff
@@ -0,0 +1,75 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+// Doc is mainly in the "int64" file :-)
+
+type (
+	// Bool is a bool which can be set/not-set and null/not-null
+	Bool struct {
+		Value bool
+		Set   bool
+		Null  bool
+	}
+	// RequiredBool must be set and not null
+	RequiredBool struct{ Bool }
+	// NullableBool must be set and can be null
+	NullableBool struct{ Bool }
+	// OptionalBool can be not set. If set, cannot be null.
+	OptionalBool struct{ Bool }
+	// OptNullBool can be not set or null
+	OptNullBool struct{ Bool }
+)
+
+// NewBool creates a Bool which is not-null and set with the given value
+func NewBool(s bool) *Bool {
```

Comment:
Building custom types and wrapping gorm with yet another, custom layer of abstraction seem like a very bad idea. I think you should operate on `bool` instead of using this.

Reply:
https://github.com/golang/go/blob/master/src/database/sql/sql.go#L266

Reply:
that's not become of that, we already discussed it. see comment below